spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -303,13 +304,13 @@ public RetryTemplate buildRetryTemplate(ConsumerProperties properties) {
 		private static ConcurrentMap<String, MimeType> mimeTypesCache = new ConcurrentHashMap<>();
 
 		static MimeType mimeTypeFromObject(Object payload, String originalContentType) {
+
 			Assert.notNull(payload, "payload object cannot be null.");
 			if (payload instanceof byte[]) {
 				return MimeTypeUtils.APPLICATION_OCTET_STREAM;
 			}
 			if (payload instanceof String) {
-				return MimeTypeUtils.APPLICATION_JSON_VALUE.equals(originalContentType) ? MimeTypeUtils.APPLICATION_JSON
-						: MimeTypeUtils.TEXT_PLAIN;
+				return originalContentType.toLowerCase().contains(MimeTypeUtils.APPLICATION_JSON.getSubtype()) ? MimeType.valueOf(originalContentType) : MimeTypeUtils.TEXT_PLAIN;
```

Comment:
@ilayaperumalg this defeats the purpose of this PR. The intention is to **not** set the contentType to `application/json` but instead respect the contentType set by the user. If we modify it, then the contentType that was `application/vnd.type.version+json` will become `application/json` and then we loose the contentType on the flow. 

Although the `originalContetType` could preserve on the first `hop` but in my opinion is **wrong** what we do: `contentType:text/plain`, `originalContentType:application/vnd.type.version+json`. 

**No**, this indicates a transformation, think about it, I'm saying the content type now is X where it used to be Y.
This PR is to make sure we honor the contentType set by the user in case of custom json types.

IMHO we should actually discuss our strategy for contentType for 2.0, I have a few pointers to share with the team